napari/_vispy/_tests/test_vispy_labels_layer.py

```diff
@@ -95,7 +99,7 @@ def test_labels_painting_with_mouse(
         position=(0, 10, 10),
         dims_displayed=(0, 1),
     )
-    visual = viewer.window._qt_viewer.layer_to_visual[layer]
+    visual = qt_viewer.layer_to_visual[layer]
```

Comment:
As above — this feels like an anti-pattern — we should have a way to provide a QtViewer that doesn't involve three separate (and repeated) lines of code?

Reply:
The antipattern is using the whole `Viewer` when just `QtViewer` is enough, as it slowdown the test and exposes the next place where something may fail. 

Having this in the fixture may be a good idea, but it is the next refactor. And I would like to spend time of proposal to SDG, so do not know when find time to do this.

Reply:
LOL Thanks! I'm still upset that pytest magic means that qt_viewer is in fact linked to the viewer_model fixture, but that's not in any way obvious from the writing of the test locally. But that's inevitable within pytest. Thanks! 🙏

Reply:
If you want, we may use `qt_viewer.viewer` attribute. As viewer_model is stored under this public attribute.